src/main/java/org/neo4j/sql2cypher/Translator.java

```diff
@@ -461,15 +448,55 @@ else if (c instanceof QOM.IsNotNull e) {
 		}
 	}
 
-	private Node node(Table<?> t) {
+	private PatternElement resolveTableOrJoin(Table<?> t) {
+		if (t instanceof QOM.Join<?> ta && ta.$on() instanceof QOM.Eq<?> eq) {
+			var relType = eq.$arg1().getQualifiedName().last();
+			Table<?> t1 = ta.$table1();
+			Table<?> t2 = ta.$table2();
+			System.out.println("t1 = " + Arrays.toString(t1.getQualifiedName().getName()));
+			System.out.println("t2 = " + Arrays.toString(t2.getQualifiedName().getName()));
+			if (/*mightBeNodeLabel(t1.$name().first()) &&*/ resolveTableOrJoin(t1) instanceof Node from
+					/*&& mightBeNodeLabel(t2.$name().first())*/ && resolveTableOrJoin(t2) instanceof Node to
+					&& mightBeRelationshipType(relType) // upper-case-column-name
+			) {
+				// todo check against primary key
+				// var targetIsId = eq.$arg2().getQualifiedName().last().equals("id");
```

Comment:
is there a better way of representing "id" of the target, not sure (also not sure how well supported something else would be in tools, like `t.ROWID` or such)